src/lib/y2network/config.rb

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "y2network/interface"
+
+module Y2Network
+  # This class represents the current network configuration including interfaces,
+  # routes, etc.
+  #
+  # @example Reading from wicked
+  #   config = Y2Network::Config.from(:sysconfig)
+  #   config.interfaces.map(&:name) #=> ["lo", eth0", "wlan0"]
+  class Config
+    # @return [Array<Interface>]
+    attr_reader :interfaces
+    # @return [Array<RoutingTable>]
+    attr_reader :routing_tables
+
+    # @param source [Symbol] Source to read the configuration from
+    class << self
+      def from(source)
```

Comment:
this is a bit confusing for me. Why we need to pass source? I expect that network config recognize itself what network daemon it uses. from is fine, but I expect here also `system` which reads what system currently use. Of course it should not be static variable, but instead read what is really on system.

Reply:
For wicked we have different sources, and that is another question, in that case that probably mean read from sysconfig + read from wicked XML and merge them. So, as you said, I'm also fine with `from`method and it is useful, we should think about where to put the responsibility about handling the configs. And not only that, we could want to merge the wicked config with the network manager config.